apps/src/lab2/views/components/Instructions/ResourcePanel/ValidationPanel.tsx

```diff
@@ -16,15 +16,19 @@ const ValidationPanel: React.FC<ValidationSettings> = ({
   return (
     <div className={validationStyles.validationPanel}>
       <div className={validationStyles.validationBubble}>
-        <ValidationTable />
-        <ValidationButton
-          onValidate={onValidate}
-          onStopValidation={onStopValidation}
-          isValidating={isValidating}
-          isValidateDisabled={isValidateDisabled}
-          buttonColor="purple"
-          buttonType="primary"
-        />
+        <div id="resource-panel-validation-table">
+          <ValidationTable />
+        </div>
+        <div id="resource-panel-validate-button">
```

Comment:
Adding button container div to ensure introjs panel can attach to element even if not yet available in dom.

Reply:
Will this add extra whitespace if there's no content in the panel or the button isn't displayed?

Reply:
Oh I meant that sometime `ValidationTable` and `ValidationButton` return null, and I wanted to make sure adding an extra div around them didn't cause any extra whitespace. It's probably not a big deal either way though!